src/NanoProtocol.cpp

```diff
@@ -0,0 +1,30 @@
+#include "NanoProtocol.h"
+#include <stdio.h>
+#include <FileIO.h>
+#include <g3log/g3log.hpp>
+NanoProtocol::NanoProtocol(const std::string& location) {
+   if (location.find("ipc") != std::string::npos) {
+      mProtocol = Protocol::IPC;
+      mLocation = location.substr(location.find("ipc://") + 6);
```

Comment:
@KjellKod 6 chars in "ipc://" we are substringing from the end of the protocol marker to the end of the location string to get the file location